components/schema-migrator/migrations/director/20231020141132_add-entity-type-and-make-part-of-products-optional-for-packages.up.sql

```diff
@@ -0,0 +1,205 @@
+    short_description           VARCHAR(256),
+    description                 TEXT,
+    system_instance_aware       BOOLEAN,
+    changelog_entries           JSONB,
+    package_id                  UUID,
+        CONSTRAINT entitytype_package_id_fk FOREIGN KEY (package_id) REFERENCES packages (id) ON DELETE CASCADE,
+    visibility                  VARCHAR(256) NOT NULL,
+    links                       JSONB,
+    part_of_products            JSONB,
+    policy_level                policy_level,
+    custom_policy_level         VARCHAR(256),
+    release_status              release_status,
+    sunset_date                 VARCHAR(256),
+    successors                  JSONB,
+    extensible                  JSONB,
+    tags                        JSONB,
+    labels                      JSONB,
+    documentation_labels        JSONB,
+    resource_hash               VARCHAR(255),
+    version_value               VARCHAR(256),
+    version_deprecated          BOOLEAN,
+    version_deprecated_since    VARCHAR(256),
+    version_for_removal         BOOLEAN
+);
+
+
+DROP VIEW IF EXISTS correlation_ids_entity_types;
+
+CREATE VIEW correlation_ids_entity_types AS
+SELECT id                  AS entity_type_id,
+       elements.value      AS value
+FROM entity_types,
+     jsonb_array_elements_text(entity_types.correlation_ids) AS elements;
+
+
+DROP VIEW IF EXISTS changelog_entries_entity_types;
+
+CREATE VIEW changelog_entries_entity_types AS
+SELECT id                      AS entity_type_id,
+       entries.version         AS version,
+       entries."releaseStatus" AS release_status,
+       entries.date            AS date,
+       entries.description     AS description,
+       entries.url             AS url
+FROM entity_types,
+     jsonb_to_recordset(entity_types.changelog_entries) AS entries(version TEXT, "releaseStatus" TEXT, date TEXT, description TEXT, url TEXT);
+
+
+DROP VIEW IF EXISTS links_entity_types;
+
+CREATE VIEW links_entity_types AS
+SELECT id                AS entity_type_id,
+       links.title       AS title,
+       links.url         AS url,
+       links.description AS description
+FROM entity_types,
+     jsonb_to_recordset(entity_types.links) AS links(title TEXT, description TEXT, url TEXT);
+
+
+DROP VIEW IF EXISTS entity_type_product;
+
+CREATE VIEW entity_type_product AS
+SELECT entity_types.id  AS entity_type_id,
+       p.id             AS product_id
+FROM entity_types,
+     jsonb_array_elements_text(entity_types.part_of_products) AS elements
+         JOIN products p ON elements.value = p.ord_id WHERE p.app_id = entity_types.app_id OR p.app_id IS NULL;
+
+DROP VIEW IF EXISTS entity_type_successors;
+
+CREATE VIEW entity_type_successors AS
+SELECT id             AS entity_type_id,
+       elements.value AS value
+FROM entity_types,
+     jsonb_array_elements_text(entity_types.successors) AS elements;
+
+
+DROP VIEW IF EXISTS entity_type_extensible;
+
+CREATE VIEW entity_type_extensible AS
```

Comment:
This view is not needed because the necessary job is done at the last line of the `tenants_entity_types` view. You can delete it.